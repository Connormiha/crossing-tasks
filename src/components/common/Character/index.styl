@keyframes decrease
    0%
        width: 32px
        height: 32px
    100%
        width: 24px
        height: 24px

// @stylint off
$characters =   'cabbage',
                'farmer',
                'sheep',
                'wolf',
                'dog',
                'caveman',
                'priest',
                'woman_black',
                'woman_white',
                'woman_red',
                'woman_blue',
                'men_black',
                'men_white',
                'men_red',
                'men_blue',
                'boy_red',
                'boy_yellow',
                'girl_red',
                'girl_yellow',
                'policeman',
                'criminal',
                'monkey',
                'gorilla'
// @stylint on

.character
    position: relative
    background-size: contain
    width: 32px
    height: 32px
    margin: 5px 0
    cursor: pointer

    &_skin
        for $item in $characters
            &_{$item}
                background-image: url(/src/assets/images/$item.svg)

        // http://www.flaticon.com/free-icon/cabbage_135237#term=cabbage&page=1&position=10
        // http://www.flaticon.com/free-icon/farmer_10613#term=farmer&page=1&position=8
        // http://www.flaticon.com/free-icon/sheep_220082#term=sheep&page=1&position=8
        // http://www.flaticon.com/free-icon/wolf_165191#term=wolf&page=1&position=17
        // http://www.flaticon.com/free-icon/caveman_78967#term=Caveman&page=1&position=1
        // http://www.flaticon.com/free-icon/priest_90522#term=Priest&page=1&position=17
        // http://www.flaticon.com/free-icon/modeling-woman_83146#term=woman%20model&page=1&position=1
        // http://www.flaticon.com/free-icon/modeling-woman_83146#term=woman%20model&page=1&position=1
        // http://www.flaticon.com/free-icon/modeling-woman_83146#term=woman%20model&page=1&position=1
        // http://www.flaticon.com/free-icon/flamenco-male-model_40045#term=men%20model&page=1&position=1
        // http://www.flaticon.com/free-icon/monkey_194978
        // http://www.flaticon.com/free-icon/gorilla-facing-right_84171

    &_packed
        width: 24px
        height: 24px
        margin: 0 auto
        animation: decrease 0.5s
