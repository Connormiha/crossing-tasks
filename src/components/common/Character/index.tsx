import style from './index.styl';

import React from 'react';
import {bindMethods} from 'helpers';

import bem from 'bem-css-modules';

const b = bem(style);

interface Props extends React.Props<any> {
    packed?: boolean;
    name: string;
    id: string;
    onClick(id: string): void;
}

export default class Character extends React.Component<Props> {
    constructor(props) {
        super(props);
        bindMethods(this, ['handleClick']);
    }

    shouldComponentUpdate() {
        return false;
    }

    handleClick() {
        this.props.onClick(this.props.id);
    }

    render() {
        let {packed, name} = this.props;

        return (
            <div className={b({skin: name.toLowerCase(), packed: !!packed})} onClick={this.handleClick} title={name} />
        );
    }
}
