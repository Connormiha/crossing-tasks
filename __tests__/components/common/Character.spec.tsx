import * as renderer from 'react-test-renderer';
import Character from 'components/common/Character';

import * as noop from 'lodash/noop';

/* tslint:disable:jsx-wrap-multiline */
describe('<Character />', () => {
    it('should render', () => {
        let tree = renderer.create(
            <Character
                name="farmer"
                onClick={noop}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render packed', () => {
        let tree = renderer.create(
            <Character
                name="sheep"
                packed
                onClick={noop}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
