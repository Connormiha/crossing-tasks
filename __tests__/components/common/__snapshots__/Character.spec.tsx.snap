// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character /> should render 1`] = `
<button
  className="character character_skin_farmer"
  onClick={[Function]}
  title="farmer"
/>
`;

exports[`<Character /> should render packed 1`] = `
<button
  className="character character_skin_sheep character_packed"
  onClick={[Function]}
  title="sheep"
/>
`;
